lib/private/AppFramework/Middleware/AdditionalScriptsMiddleware.php

```diff
@@ -42,6 +44,14 @@ public function __construct(EventDispatcherInterface $dispatcher, IUserSession $
 	}
 
 	public function afterController($controller, $methodName, Response $response): Response {
+		/*
+		 * There is no need to emit these signals on a public share page
+		 * There is a separate event for that already
+		 */
+		if ($controller instanceof PublicShareController || $controller instanceof StandaloneTemplateResponse) {
```

Comment:
```suggestion
		if ($controller instanceof PublicShareController || $response instanceof StandaloneTemplateResponse) {
```

Or did I miss something?

Reply:
It of course helps if I push the latest changes :S sorry.
Anyway fixed and with tests